execution.go

```diff
@@ -28,7 +29,62 @@ func SetupExecutionEnvironment() {
 	fset = token.NewFileSet()
 }
 
-// OutputMsg holds the data for a pyout message.
+type KernelStatusMsg struct {
+	ExecState string `json:"execution_state"`
+}
+
+func setBusy(receipt MsgReceipt) {
+	var status KernelStatusMsg
+
+	msg := NewMsg("status", receipt.Msg)
+	status.ExecState = "busy"
+	msg.Content = status
+
+	receipt.SendResponse(receipt.Sockets.IOPubSocket, msg)
+}
+
+func setIdle(receipt MsgReceipt) {
+	var status KernelStatusMsg
+
+	msg := NewMsg("status", receipt.Msg)
+	status.ExecState = "idle"
+	msg.Content = status
+
+	receipt.SendResponse(receipt.Sockets.IOPubSocket, msg)
+}
+
+
+type ExecuteInputMsg struct {
+	ExecCount int    `json:"execution_count"`
+	Code      string `json:"code"`
+}
+
+func publishExecuteInput(receipt MsgReceipt, execCount int, code string) {
+	var pubExecInputContent ExecuteInputMsg
+	pubExecInput := NewMsg("execute_input", receipt.Msg)
+	pubExecInputContent.ExecCount = execCount
+	pubExecInputContent.Code = code
+	receipt.SendResponse(receipt.Sockets.IOPubSocket, pubExecInput)
+}
+
+type StreamMsg struct {
+	Stream string `json:"name"`
+	Data string `json:"text"`
+}
+func publishWriteStream(receipt MsgReceipt, stdout bool, text string) {
+	var pubWriteStreamContent StreamMsg
```

Comment:
same comment than for `publishExecuteInput`.
it looks like there is a typo, or a line like the following that is missing:

```go
pubWriteStream.Content = pubWriteStreamContent
```

if that's the case, I would just rename `pubWriteStreamContent` into `content`.